src/ucp/core/ucp_request.h

```diff
@@ -205,7 +205,10 @@ struct ucp_request {
                             } header UCS_S_PACKED; /* packed to avoid 32-bit
                                                       padding */
                             uint16_t       am_id;
-                            uint16_t       flags;
+                            /* API flags from @ref ucp_send_am_flags enum */
+                            uint8_t        flags: 7;
+                            /* Whether header is sent already */
+                            uint8_t        hdr_sent: 1;
```

Comment:
imo, this name is more confusing than it was before. With AM API we send headers partially and user header goes to the last fragment. Also, previously we avoided using bit fields, because compilers tend to generate more instructions with them and these fields are accessed on the fast path

Reply:
As we checked together, user header still goes into the first fragment
I agree that the bit field may require some extra cycles, that's why I didn't have it initially.
Then let's go back to one extra field `uint8_t internal_flags`.
The size of request should not grow, because `rndv` part of the union is significantly larger anyway
@yosefe ?

Reply:
yes, i was wrong about user header going to the last fragment. It goes to the end of the first fragment

Reply:
if we want to avoid bitmap, we can define internal AM flag that goes into "flags" field, and does not collide with the API flags

Reply:
ok, then I add back the internal enum, with new internal flag using bit 15:
```C
enum ucp_request_am_internal_flags {
    UCP_REQUEST_AM_FLAG_HEADER_SENT = UCS_BIT(15)
};
```
Or we could add one last element to `enum ucp_send_am_flags {`
```C
enum ucp_send_am_flags {
    UCP_AM_SEND_FLAG_REPLY       = UCS_BIT(0),
    UCP_AM_SEND_FLAG_EAGER       = UCS_BIT(1),
    UCP_AM_SEND_FLAG_RNDV        = UCS_BIT(2),
    UCP_AM_SEND_FLAG_COPY_HEADER = UCS_BIT(3),
    UCP_AM_SEND_FLAG_LAST  << And use this in the internal enum
```

Reply:
reverted to separate `uint8_t internal_flags`